scriptmodules/packages.sh

```diff
@@ -386,45 +440,197 @@ function rp_hasBinary() {
+                fi
+            fi
+            [[ -z "$ret" ]] && ret=2
+            ;;
+        source)
+            local repo_type="${__mod_info[$id/repo_type]}"
+            local repo_url="$(rp_resolveRepoParam "${__mod_info[$id/repo_url]}")"
+            case "$repo_type" in
+                file)
+                    if [[ -n "$pkg_repo_date" ]]; then
+                        local file_date="$(rp_getFileDate "$repo_url")"
+                        rp_dateIsNewer "$pkg_repo_date" "$file_date"
+                        ret="$?"
+                    fi
+                    ;;
+                git|svn)
+                    local repo_branch="$(rp_resolveRepoParam "${__mod_info[$id/repo_branch]}")"
+                    local repo_commit="$(rp_resolveRepoParam "${__mod_info[$id/repo_commit]}")"
+                    # if we are locked to a single commit, then we compare against the current module commit only
+                    if [[ -n "$repo_commit" ]]; then
+                        # if we are using git and the module has an 8 character commit hash, then adjust
+                        # the package commit to 8 characters also for the comparison
+                        if [[ "$repo_type" == "git" && "${#repo_commit}" -eq 8 ]]; then
+                            pkg_repo_commit="${pkg_repo_commit::8}"
+                        fi
+                        if [[ "$pkg_repo_commit" != "$repo_commit" ]]; then
+                            ret=0
+                        fi
+                    else
+                        local remote_commit
+                        if remote_commit="$(rp_getRemoteRepoHash "$repo_type" "$repo_url" "$repo_branch" "$repo_commit")"; then
+                            if [[ -n "$remote_commit" && "$pkg_repo_commit" != "$remote_commit" ]]; then
+                                ret=0
+                            fi
+                        else
+                            __ERRMSGS+=("$remote_commit")
+                            ret=3
+                        fi
+                    fi
+                    ;;
+                :*)
+                    local pkg_repo_extra="${__mod_info[$id/pkg_repo_extra]}"
+                    # handle checking via module function - function should return 0 if there is a newer version
+                    local function="${repo_type:1}"
+                    if fnExists "$function"; then
+                        "$function" newer
+                        ret="$?"
+                    fi
+                    ;;
+                *)
+                    # fallback on forcing an update
+                    ret=2
+                    ;;
+            esac
+
+            # if we are currently not going to update - check the date of the module code
+            # if it's newer than the install date of the module we force an update
+            if [[ "$ret" -eq 1 && "$__ignore_module_date" -ne 1 ]]; then
+                local module_date="$(date -Iseconds -r "${__mod_info[$id/path]}")"
```

Comment:
Is this correct ? `module_date` shouldn't be using `$__mod_info[$id/date]` ?

Reply:
It compares against the last change date of the script/module file. So if a scriptmodule is changed an update is forced.